### PR TITLE
feat(fhir): add Encounter and Procedure R4 resource generators (#387)

### LINKED ISSUE #387: feat(fhir): add Encounter and Procedure FHIR R4 resource generators
## Context
The FHIR gateway has generators for Patient, Observation, Condition, MedicationStatement, and AllergyIntolerance but is missing Encounter and Procedure. Both DB tables already exist:

- `encounters` table: encounter_type, status, start_time, end_time, provider_id, location, reason, notes
- `procedures` table: cpt_code, icd10_codes, status, performed_at, performed_by, notes

### Encounter Generator (`generators/encounter.ts`)
- Map `encounter_type` (inpatient, outpatient, emergency, telehealth) → FHIR Encounter.class using ActCode system
- Map internal status → FHIR EncounterStatus (planned, arrived, in-progress, finished, cancelled)
- Include period (start_time/end_time), participant[] (provider_id), location, reasonCode
- Reference subject (Patient) and participant (Practitioner)

### Procedure Generator (`generators/procedure.ts`)
- Map CPT codes → FHIR Procedure.code with `http://www.ama-assn.org/go/cpt` system
- Map ICD-10 codes → reasonCode
- Map internal status → FHIR ProcedureStatus
- Include performedDateTime, performer[] (performed_by), subject reference

### Integration
- Export from `generators/index.ts`
- Include in `exportPatient` bundle in `router.ts` (add encounters + procedures to the parallel fetch)
- Add to existing FHIR export test coverage

## Acceptance criteria
- `toFhirEncounter()` and `toFhirProcedure()` functions exported
- Both included in patient FHIR bundle export
- Unit tests for mapping logic (status, coding, edge cases)
- TypeScript strict, ESM imports

## Dependencies
- #386 (FHIR R4 type definitions)

### PR DESCRIPTION
**Pull request by blamechris**
## Summary

- `services/fhir-gateway/src/generators/encounter.ts` — maps `encounter_type` to HL7 v3 ActCode (inpatient → IMP, outpatient → AMB, emergency → EMER, telehealth → VR, home → HH, observation → OBSENC). Status passes through when it matches the FHIR EncounterStatus set, otherwise \"unknown\". Emits period, participant (Practitioner via `provider_id`), reasonCode (text from `reason`)
- `services/fhir-gateway/src/generators/procedure.ts` — CPT codes attach to `Procedure.code` via `http://www.ama-assn.org/go/cpt`; ICD-10 codes become `reasonCode` entries. Status remaps internal variants (\"scheduled\" → preparation, \"cancelled\" → stopped) so downstream FHIR consumers only see spec-valid values
- `exportPatient` in `router.ts`: added encounters + procedures to the parallel fetch and bundle assembly
- Updated existing export-audit / headers / purge-warning test mocks with the two new table handles
- 49 new generator tests (26 encounter + 23 procedure) covering status boundaries, class-code mapping, optional-field omissions, and ICD-10 `reasonCode` edge cases

## Test plan

- [x] `pnpm --filter @carebridge/fhir-gateway test` — 129/129
- [x] `pnpm typecheck` — 40/40
- [x] All existing export tests still pass after the parallel-fetch addition

Closes #387

### CODE REVIEW

**Comment by blamechris** [comment]
## CareBridge Inspector review

**Verdict:** Approve with follow-ups

Solid FHIR R4 work. System URLs, value-set handling, reference formats, and status remaps are all spec-correct. Test coverage is thorough (status boundaries, class mapping, optional-field omissions, ICD-10 edge cases), and the parallel-fetch + test-mock updates match the existing pattern. Zero Attribution clean.

### FHIR correctness — verified

- `Encounter.class.system` = `http://terminology.hl7.org/CodeSystem/v3-ActCode`, codes IMP/AMB/EMER/VR/HH/OBSENC are all valid v3 ActCode values
- `Procedure.code` CPT system URL `http://www.ama-assn.org/go/cpt` and ICD-10 `http://hl7.org/fhir/sid/icd-10-cm` are both correct per FHIR R4
- `ProcedureStatus` remap (`scheduled` → `preparation`, `cancelled` → `stopped`) matches the R4 value set
- `EncounterStatus` pass-through with `unknown` fallback is correct
- Reference format (`Patient/…`, `Practitioner/…`) is correct, `period.end` correctly omitted for in-progress encounters

### Observations (non-blocking)

1. **`encounter.notes` / `procedure.notes` not emitted** — both are `encryptedText` in the DB. Correct call to omit from patient-facing FHIR bundles; internal clinician narrative isn't the same as a formal FHIR text narrative. Worth a code comment noting this is intentional, but not required.

2. **`Encounter.location` not emitted** — the `encounters.location: text` column exists in the DB but the generator ignores it. Tracked as follow-up. See https://github.com/blamechris/carebridge/issues/937.

3. **Unknown `encounter_type` silently falls back to `AMB`** — the test `falls back to AMB for unknown encounter_type` locks this in, but silently defaulting an unknown class to ambulatory can mask data-quality issues. Consider either (a) logging a warning via `@carebridge/logger` when the fallback fires, or (b) mapping unknown to the R4 `class` nullFlavor coding. Not blocking; flagging for a future follow-up.

4. **`scheduled` → `preparation` semantic mismatch** — FHIR `preparation` means \"the patient is being prepared right now\" while our internal `scheduled` usually means \"set for a future date\". There's no better R4 mapping, so this is acceptable. Worth a comment in `mapProcedureStatus` to flag the known semantic drift for future maintainers.

5. **No `meta.profile` on exported resources** — if/when CareBridge claims US Core conformance, all exported resources will need profile URLs. Cross-cutting across every generator. Tracked as follow-up. See https://github.com/blamechris/carebridge/issues/938.

6. **CPT licensing (compliance, not code)** — AMA licenses CPT code distribution. External FHIR bundle recipients may trigger a license requirement. Not a PR concern. See https://github.com/blamechris/carebridge/issues/939.

### Findings

0 critical, 3 suggestions (comments for #1 / #3 / #4), 3 deferred follow-ups (issues #937, #938, #939).

**Rebase required:** No — 49 new tests pass, the three existing export-test mock updates are the only touches to shared files.

Nice, clean landing for #387.

File: services/fhir-gateway/src/__tests__/export-purge-warning.test.ts [hunk 1]
```diff
@@ -17,6 +17,8 @@ const labResultsTable = { __name: "lab_results" };
 const medicationsTable = { __name: "medications" };
 const diagnosesTable = { __name: "diagnoses" };
 const allergiesTable = { __name: "allergies" };
+const encountersTable = { __name: "encounters" };
+const proceduresTable = { __name: "procedures" };
```
[the remark this answers is not in the document]
Comment (blamechris) [reply services/fhir-gateway/src/__tests__/export-purge-warning.test.ts:21]: **FIX**

Fixed in `9531492`. `toFhirEncounter` and `toFhirProcedure` stubs added to the generators mock in `export-purge-warning.test.ts`.

File: services/fhir-gateway/src/generators/encounter.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        individual: {
+          reference: `Practitioner/${encounter.provider_id}`,
+        },
+      },
+    ];
+  }
+
+  if (encounter.reason) {
+    resource.reasonCode = [{ text: encounter.reason }];
+  }
```
[the remark this answers is not in the document]
Comment (blamechris) [reply services/fhir-gateway/src/generators/encounter.ts:110]: **FIX**

Fixed in `9531492`. `FhirEncounter` type gains optional `location: EncounterLocation[]`; `toFhirEncounter` maps the internal `location` column to `Encounter.location[].location.display`. The `Reference` type also grew an optional `display` field so we can surface the free-text location without inventing a Location resource id. When a Location resource set exists the emit can switch to `reference: \"Location/<id>\"`.

```diff
  if (encounter.reason) {
    resource.reasonCode = [{ text: encounter.reason }];
  }

+ if (encounter.location) {
+   resource.location = [
+     { location: { display: encounter.location } },
+   ];
+ }
  return resource;
```

Two new tests lock it in — emit when location is set, omit when null.

File: services/fhir-gateway/src/__tests__/export-bundle-audit.test.ts [hunk 3]
```diff
@@ -25,6 +25,8 @@ const labResultsTable = { __name: "lab_results" };
 const medicationsTable = { __name: "medications" };
 const diagnosesTable = { __name: "diagnoses" };
 const allergiesTable = { __name: "allergies" };
+const encountersTable = { __name: "encounters" };
+const proceduresTable = { __name: "procedures" };
```
[the remark this answers is not in the document]
Comment (blamechris) [reply services/fhir-gateway/src/__tests__/export-bundle-audit.test.ts:29]: **FIX**

Fixed in `9531492`. Same generators-mock stub applied to `export-bundle-audit.test.ts`.

File: services/fhir-gateway/src/__tests__/export-headers.test.ts [hunk 4]
```diff
@@ -17,6 +17,8 @@ const labResultsTable = { __name: "lab_results" };
 const medicationsTable = { __name: "medications" };
 const diagnosesTable = { __name: "diagnoses" };
 const allergiesTable = { __name: "allergies" };
+const encountersTable = { __name: "encounters" };
+const proceduresTable = { __name: "procedures" };
```
[the remark this answers is not in the document]
Comment (blamechris) [reply services/fhir-gateway/src/__tests__/export-headers.test.ts:21]: **FIX**

Fixed in `9531492`. Same generators-mock stub applied to `export-headers.test.ts`.

**Comment by blamechris** [comment]
## Review Comments Addressed

| # | Comment | Outcome | Reference |
|---|---------|---------|-----------|
| 1 | export-purge-warning.test.ts generators mock missing Encounter/Procedure stubs | FIX | `9531492` |
| 2 | `encounter.location` column silently dropped | FIX | `9531492` |
| 3 | export-bundle-audit.test.ts generators mock missing stubs | FIX | `9531492` |
| 4 | export-headers.test.ts generators mock missing stubs | FIX | `9531492` |

**Total:** 4 comments addressed
- Fixed: 4 (`9531492`)
- False positives: 0
- Follow-up issues: 0 (earlier agent-review follow-ups #937, #938, #939 remain open for US Core profile conformance and compliance work)
